src/styles/search-bar.css

```diff
@@ -0,0 +1,111 @@
+/* #region search-bar */
+.search-bar {
+  font-family: Avenir, sans-serif;
+  font-weight: 300;
+  color: var(--font-color);
+
+  margin: 20px auto 0;
+
+  border: 1px solid var(--border-color);
+  border-radius: 4px;
+  box-shadow: 0 1px 8px 0 var(--shadow-color);
+  background-color: var(--background-color);
+}
+
+.search-bar:hover {
+  box-shadow: 0 4px 4px 0 var(--shadow-color-2);
+}
+
+.search-bar:has(.search-bar__content-wrapper__input:focus) {
+  background: linear-gradient(
+    180deg,
+    var(--background-color) 0%,
+    var(--background-color-2) 100%
+  );
+  font-weight: 600;
+}
```

Comment:
The `:has()` pseudo-class is not widely supported across all browsers. If you need to ensure compatibility with older browsers, consider using JavaScript to achieve similar functionality.